src/quo2/theme.cljs

```diff
@@ -22,3 +25,31 @@
   ([light-value dark-value override-theme]
    (let [theme (or override-theme (get-theme))]
      (if (= theme :light) light-value dark-value))))
+
+(defn provider
+  "Wrap `children` in a React Provider using `quo2.theme/theme-context` as the
+  context.
+
+  `options`: Clojure map. Currently we only use the `:theme` key. In the future
+  we may support other settings.
+  "
+  [options & children]
+  (into [:> (.-Provider theme-context) {:value (clj->js options)}]
+        children))
+
+(defn use-theme
+  "A hook that returns the current theme context."
+  []
+  (utils.transforms/js->clj (rn/use-context theme-context)))
```

Comment:
It is necessary if we want `use-theme` to return a Clojure data structure, which I often prefer in order to avoid "leaking" JS objects to the outside world. Otherwise callers of `use-theme` need to use the macro `oget`, which I think is more error prone than just core Clojure functions.

From a performance standpoint, I think it makes no measurable difference, but of course `oget` is slightly faster without the data conversion of `js->clj`.

Reply:
i mean value could be returned i guess

Reply:
Yes, it could be @flexsurfer, but then any other place in the code that calls `use-theme` will need to deal with a JS object. That's the trade-off in maintainability IMO. Right now the only place calling `use-theme` is `f-with-theme`, but this may change in the future, and then we will have more places dealing with JS objects.

For this PR both approaches will do :)

Reply:
> but then any other place in the code that calls `use-theme` will need to deal with a JS object.

why? it will return a keyword

Reply:
We are storing a map in the context when `provider` is used (e.g. `[provider {:theme :dark} ...]`). When using `use-theme`, if we want to return the original value stored in `theme-context`, we need to convert the value back to CLJ using `js->clj`, otherwise it's a JS object like `#js {:theme "dark"}`.

In the future, we could store more things inside `theme-context` as the app becomes more "themeable", that's why @J-Son89 and I chose to use a map such as `{:theme :dark}` instead of the keyword `:dark`.

For this PR we can do basically anything because there's only one usage of `use-theme`, but the future proof approach IMO is to store a map in the context and retrieve a CLJ map when `use-theme` is called, not a keyword.